lib/cursor.js

```diff
@@ -358,7 +360,7 @@ Cursor.prototype._killcursor = function(callback) {
   this.cursorState.dead = true;
   this.cursorState.killed = true;
   // Remove documents
-  this.cursorState.documents = [];
+  clearCursorDocuments(this);
```

Comment:
There is one slight difference between the original code and what I have hear in that `clearCursorDocuments` will reset both `cursorState.documents` AND `cursorState.cursorIndex`. The original code did not reset the `cursorIndex`.  This seems like a trivial difference, but I thought I'd point it out.